### PR TITLE
Fix/kg batch size issue

### PR DESCRIPTION
**Pull request by IGZVasjaUrbancic**
A fix for the issue caused by padding in batches

### CODE REVIEW

File: kg/pipe/compute_vectors.py [hunk 1]
```diff
@@ -16,8 +16,7 @@
 DATASET_PATH = ASSET_PATH + "dataset.pkl.gz"
 
 # Constants
-VECTOR_DIMS = 768
-BATCH_SIZE = 1  # This batch size is a workaround for a padding problem.
+BATCH_SIZE = 1    # This batch size is a workaround for a padding problem.
```
Comment (marcvivancos) [inline kg/pipe/compute_vectors.py:19]: Take this comment out now that this works

File: kg/server/recommender.py [hunk 2]
```diff
@@ -62,12 +62,14 @@ def _compute_query_vector(self, raw_query_text: str) -> torch.Tensor:
         Computes a vector for a given query input.
         """
         self.log.debug(f"Computing vector for query text ('{raw_query_text[:120]}...'")
+        device = self.model.device
```
Comment (marcvivancos) [inline kg/server/recommender.py:65]: Does this work for cpu/gpu, isn't this redundant if we do not load the model to a device?

File: kg/server/recommender.py [hunk 3]
```diff
@@ -62,12 +62,14 @@ def _compute_query_vector(self, raw_query_text: str) -> torch.Tensor:
...
 
         processed_query = self._preprocess(raw_query_text)
         tokens = self._tokenize(processed_query)
 
-        query_token_vecs = self.model(**tokens)[0].detach().squeeze()
-        query_vector = torch.mean(query_token_vecs, dim=0)
+        query_token_vecs = self.model(input_ids=tokens['input_ids'].to(device),
+                                      attention_mask=tokens['attention_mask'].to(device))[0].detach().squeeze()
+        query_vector = torch.mean(query_token_vecs, dim=0).cpu()
```
Comment (marcvivancos) [inline kg/server/recommender.py:72]: Same as above with making the device explicit

**Review by marcvivancos — APPROVED** [review]
